### PR TITLE
Support for redis sentinel in redis input and output

### PR DESCRIPTION
**Pull request by juusom**
Redis supports failover through a separate service called Sentinel (http://redis.io/topics/sentinel). I added support for it to the output and input modules using the redis-sentinel wrapper library (https://github.com/flyerhzm/redis-sentinel).

### CODE REVIEW

**Comment by amaltson** [comment]
:+1:

File: lib/logstash/inputs/redis.rb [hunk 1]
```diff
@@ -31,6 +31,15 @@ class LogStash::Inputs::Redis < LogStash::Inputs::Threadable
   # The port to connect on.
   config :port, :validate => :number, :default => 6379
 
+  # The hostnames of Redis sentinels (overrides host)
+  config :sentinel_hosts, :validate => :array
```
Comment (jordansissel) [inline lib/logstash/inputs/redis.rb]: > The current host variable is validated as a string

Doing `:validate => :array` will accept single-value items:

```
redis {
  host => "host"
  # Also valid and means the same thing:
  host => [ "host" ]
  ...
}
```

**Comment by jordansissel** [comment]
If the complexity of the configuration settings cannot be reduced, I wonder if this shouldn't just be a separate plugin. 

My main concern is that users without sentinel will be confused by the settings, and further that we'll have to implement extra validation to ensure mutual exclusion (did you set 'hosts' and also 'sentinel_hosts' ? etc) that could be also confusing.

Thoughts?

Also, we recently moved individual plugins to individual repositories, so if you can, please split this PR and file separately for https://github.com/logstash-plugins/logstash-input-redis and https://github.com/logstash-plugins/logstash-output-redis?

**Comment by suyograo** [comment]
please move this patch to logstash-plugins (input and output)

**Pull request closed without merging**